### PR TITLE
fix(EventStack): fix erroneous removal of non-empty EventPool

### LINKED ISSUE #2905: Dropdowns misbehave when inside Popups
### Steps
1. Place a `<Dropdown />` inside a `<Popup />`, with a `<Button />` as trigger
2. Click the button (pop-up should open)
3. Click outside the pop-up (it should close itself)
4. Open the pop-up again
5. Open the dropdown
6. Click outside the pop-up

### Expected Result
The pop-up should close.

### Actual Result
The pop-up stays open. Only the dropdown closes. Clicking again outside will leave the pop-up open. The only way to close it is to click the button again.

### Version
0.81.1

### Testcase
https://codepen.io/anon/pen/QxvpNe

### LINKED ISSUE #2921: Dropdown: do not close when clicked in succession
### Steps
Please see https://codesandbox.io/s/42063j6q39

1. Click Messages 1, click messages 2, click messages 1 again. Messages 2 Dropdown will be stuck open.
2. Change semantic-ui-react version to 0.80.2, repeat step 1, bug cannot be reproduced.
3. Change semantic-ui-react version back to 0.81.1, repeat step 1, bug can again be reproduced.

### Expected Result
Dropdown menus close when clicking on another.

### Actual Result
Open dropdown menu get stuck open on version 0.81.1

### Version
0.81.1 is definitely affected.
0.80.2 is not affected.

Please note - 0.81.0 doesn't seem to work in the sandbox - not sure if this version was ever released to npm. 

The regression is somewhere between 0.80.2 and 0.81.1 releases.

### Testcase
[Fork, update, and replace this pen to show the bug]:
https://codesandbox.io/s/42063j6q39

### PR DESCRIPTION
**Pull request by mic4ael**
That fix avoids destruction of an `EventPool` in some cases which results in loss of some of the registered handlers.

Fixes #2905.
Fixes #2921.

### CODE REVIEW

**Comment by layershifter** [comment]
I will review it today, need to test it.

**Comment by layershifter** [comment]
@mic4ael does this fix an any existing issue?

**Comment by mic4ael** [comment]
@layershifter yes, Closes #2905

**Comment by layershifter** [comment]
@mic4ael very cool ❤️

File: src/lib/eventStack/EventPool.js [hunk 1]
```diff
@@ -53,13 +53,10 @@ export default class EventPool {
...
+   * @return {Boolean}
    */
-  hasHandlers(eventType) {
-    const handlerSet = this.handlerSets.get(eventType)
-
-    if (handlerSet) return handlerSet.hasHandlers()
-    return false
+  hasHandlers() {
+    return this.handlerSets.size > 0
   }
```
Comment (layershifter) [inline src/lib/eventStack/EventPool.js:60]: `EventPool` has handlers if it has any `EventHandlerSet`.

File: src/lib/eventStack/EventTarget.js [hunk 2]
```diff
@@ -49,14 +49,14 @@ export default class EventTarget {
...
-      if (newPool.hasHandlers(eventType)) {
-        this.removeTargetHandler(eventType)
+      if (newPool.hasHandlers()) {
         this.pools.set(poolName, newPool)
       } else {
-        this.removeTargetHandler(eventType)
         this.pools.delete(poolName)
       }
 
+      this.removeTargetHandler(eventType)
```
Comment (layershifter) [inline src/lib/eventStack/EventTarget.js:58]: We will need always to call `removeTargetHandler()`.

File: src/lib/eventStack/EventTarget.js [hunk 3]
```diff
@@ -49,14 +49,14 @@ export default class EventTarget {
     if (pool) {
       const newPool = pool.removeHandlers(eventType, eventHandlers)
 
-      if (newPool.hasHandlers(eventType)) {
-        this.removeTargetHandler(eventType)
+      if (newPool.hasHandlers()) {
         this.pools.set(poolName, newPool)
       } else {
-        this.removeTargetHandler(eventType)
         this.pools.delete(poolName)
```
Comment (layershifter) [inline src/lib/eventStack/EventTarget.js:55]: `EventPool` should not added if it's empty

**Comment by layershifter** [comment]
@mic4ael I added some changes, can you please confirm that everything is okay?

**Comment by mic4ael** [comment]
@layershifter yup, works fine ;)

**Comment by mic4ael** [comment]
@layershifter  what is going on with the codecov/patch job? It has been stuck for quite some time

**Comment by layershifter** [comment]
Sometimes it fails 🤔 Levi will merge this PR when he will have time.

**Comment by levithomason** [comment]
Really appreciate this fix.  I know it took some real effort to grok the problem and solve the issue ❤️

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.82.1`.

**Pull request merged**